programs/stake/src/stake_state.rs

```diff
@@ -414,29 +428,57 @@ impl Stake {
                 0
             };
 
-            total_rewards +=
-                (self.delegation.stake(*epoch, stake_history) * epoch_credits) as f64 * point_value;
```

Comment:
nits: Also, it'd be great if we could add a tiny unit-test causing this actually to overflow to avoid any future regressions. :)

Reply:
done